bentoml/yatai/deployment/aws_ec2/operator.py

```diff
@@ -0,0 +1,556 @@
+
+            logger.info("Building service")
+            build_template(
+                template_file_path, project_path, aws_ec2_deployment_config.region
+            )
+
+            logger.info("Packaging service")
+            package_template(
+                s3_bucket_name, project_path, aws_ec2_deployment_config.region
+            )
+
+            logger.info("Deploying service")
+            deploy_template(
+                deployment_stack_name,
+                s3_bucket_name,
+                project_path,
+                aws_ec2_deployment_config.region,
+            )
+
+    def add(self, deployment_pb):
+        try:
+            deployment_spec = deployment_pb.spec
+            deployment_spec.aws_ec2_operator_config.region = (
+                deployment_spec.aws_ec2_operator_config.region
+                or get_default_aws_region()
+            )
+            if not deployment_spec.aws_ec2_operator_config.region:
+                raise InvalidArgument("AWS region is missing")
+
+            ensure_sam_available_or_raise()
+            ensure_docker_available_or_raise()
+
+            bento_pb = self.yatai_service.GetBento(
+                GetBentoRequest(
+                    bento_name=deployment_spec.bento_name,
+                    bento_version=deployment_spec.bento_version,
+                )
+            )
+
+            if bento_pb.bento.uri.type not in (BentoUri.LOCAL, BentoUri.S3):
+                raise BentoMLException(
+                    "BentoML currently not support {} repository".format(
+                        BentoUri.StorageType.Name(bento_pb.bento.uri.type)
+                    )
+                )
+            bento_path = bento_pb.bento.uri.uri
+
+            return self._add(deployment_pb, bento_pb, bento_path)
+        except BentoMLException as error:
+            deployment_pb.state.state = DeploymentState.ERROR
+            deployment_pb.state.error_message = f"Error: {str(error)}"
+            return ApplyDeploymentResponse(
+                status=error.status_proto, deployment=deployment_pb
+            )
+
+    def _add(self, deployment_pb, bento_pb, bento_path):
+        try:
+            if loader._is_remote_path(bento_path):
+                with loader._resolve_remote_bundle_path(bento_path) as local_path:
+                    return self._add(deployment_pb, bento_pb, local_path)
+
+            deployment_spec = deployment_pb.spec
+            aws_ec2_deployment_config = deployment_spec.aws_ec2_operator_config
+
+            artifact_s3_bucket_name = generate_aws_compatible_string(
+                "btml-{namespace}-{name}-{random_string}".format(
+                    namespace=deployment_pb.namespace,
+                    name=deployment_pb.name,
+                    random_string=uuid4().hex[:6].lower(),
+                )
+            )
+            create_s3_bucket_if_not_exists(
+                artifact_s3_bucket_name, aws_ec2_deployment_config.region
+            )
```

Comment:
Since for the EC2 deployments, the only stored in the s3 bucket is the cloudformation configuration file, we can create a bucket that store all of the configuration file instead of creating a bucket everytime we are creating a deployment.

Reply:
How do you suggest to do that?
How can we make unique and repeatable bucket name?

Reply:
Using aws user id for creating bucket name.